crates/proof-of-sql/src/sql/proof_plans/group_by_exec.rs

```diff
@@ -288,16 +295,23 @@ impl ProverEvaluate for GroupByExec {
 
         // 4. Tally results
         let sum_result_columns_iter = sum_result_columns.iter().map(|col| Column::Scalar(col));
-        let res = group_by_result_columns
+        let columns = group_by_result_columns
             .clone()
             .into_iter()
             .chain(sum_result_columns_iter)
-            .chain(core::iter::once(Column::BigInt(count_column)))
+            .chain(iter::once(Column::BigInt(count_column)))
             .collect::<Vec<_>>();
+        let res = Table::<'a, S>::try_from_iter(
+            self.get_column_result_fields()
+                .into_iter()
+                .map(|field| field.name())
+                .zip(columns.clone().into_iter()),
```

Comment:
NIT:
We shouldn't need `clone`s/`collect`s.
```suggestion
            .chain(iter::once(Column::BigInt(count_column)));
        let res = Table::<'a, S>::try_from_iter(
            self.get_column_result_fields()
                .into_iter()
                .map(|field| field.name())
                .zip(columns),
```